[BUG]: Signal label broken in designer

### Provide a brief description of the bug.

Signal Label is broken in designer, after changing device, this is the error:

```python
BECWidgets | 2025-07-22 19:34:14 | bec_widgets.utils.error_popups | [ERROR] | SafeProperty error in SETTER of 'bec_widgets.widgets.utility.signal_label.signal_label.SignalLabel.device':
Traceback (most recent call last):
  File "/Users/janwyzula/PSI/bec_widgets/bec_widgets/utils/error_popups.py", line 67, in safe_setter
    return setter_func(self_, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/janwyzula/PSI/bec_widgets/bec_widgets/widgets/utility/signal_label/signal_label.py", line 332, in device
    self.connect_device()
  File "/Users/janwyzula/PSI/bec_widgets/bec_widgets/widgets/utility/signal_label/signal_label.py", line 256, in connect_device
    self._signal_key, self._signal_info = self._signal_key_and_info()
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/janwyzula/PSI/bec_widgets/bec_widgets/widgets/utility/signal_label/signal_label.py", line 303, in _signal_key_and_info
    signal_info = self._device_obj._info["signals"][self._signal]
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: None


```

### Describe what you expected to happen and what actually happened.

Change the device

### Outline the steps that lead to the bug's occurrence. Be specific and provide a clear sequence of actions.

Go to designer and try to change the device there

### bec_widgets version

2.30

### bec core version

3.52.1

### Any extra info / data? e.g. log output...

_No response_

### Related issues

_No response_

--- a/bec_widgets/widgets/utility/signal_label/signal_label.py
+++ b/bec_widgets/widgets/utility/signal_label/signal_label.py
@@ -300,7 +300,10 @@ class SignalLabel(BECWidget, QWidget):
 
     def _signal_key_and_info(self) -> tuple[str, dict]:
         if isinstance(self._device_obj, Device):
-            signal_info = self._device_obj._info["signals"][self._signal]
+            try:
+                signal_info = self._device_obj._info["signals"][self._signal]
+            except KeyError:
+                return "", {}
             if signal_info["kind_str"] == Kind.hinted.name:
                 return signal_info["obj_name"], signal_info
             else:
@@ -308,15 +311,6 @@ class SignalLabel(BECWidget, QWidget):
         elif isinstance(self._device_obj, Signal):
             return self._device, self._device_obj._info["describe_configuration"]
         return "", {}
-
-        # if self.dev[self._device]._info["signals"] == {}:
-        #     return self._signal or self._device
-        # signal_info = self.dev[self._device]._info["signals"][self._signal]
-        # return (
-        #     signal_info["obj_name"]
-        #     if signal_info["kind_str"] == Kind.hinted.name
-        #     else (self._signal or self._device)
-        # )
 
     @SafeProperty(str)
     def device(self) -> str:

